app-text/cherrytree: update and enable tests

goatshriek opened the pull request:
The tests for cherrytree were restricted, with a comment that dependencies were not available in Gentoo. Upon further investigation and with confirmation from the upstream maintainer (giuspen/cherrytree#2645), the test dependencies have since changed, and now everything needed to run the tests is available. This change updates the dependencies and enables the tests, excluding some that fail in the ebuild sandbox but pass when run by themselves.

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

thesamesam commented:
Thank you! Would you mind making the commit message reflect the PR description for posterity?

goatshriek commented:
My apologies, I think the commit message got dropped when I squashed the multiprocessing fix. I've updated it slightly as I've since identified the reason for the test failures and corrected it with a manual override of the job count.
